### DISCUSSION TITLE
wgpu_hal problem

### DISCUSSION THREAD

**Discussion by FALLfield** [bevyengine/bevy#14531] [Q&A]
When I am running the example of 2D shape, there are some errors:
`ERROR wgpu_hal::auxil::dxgi::exception: ID3D12CommandQueue::ExecuteCommandLists: Using ClearRenderTargetView on Command 
List (0x000001EB6DBDB290:'Unnamed ID3D12GraphicsCommandList Object'): Resource state (0xBF53EAF0: D3D12_RESOURCE_STATE_[COMMON|PRESENT]) of resource
 (0x000001EB6DAEDFD0:'Unnamed ID3D12Resource Object') (subresource: 0) is invalid for use as a render target.  Expected State Bits (all): 0xBF53EAD0
: D3D12_RESOURCE_STATE_RENDER_TARGET, Actual State: 0xBF53EAB0: D3D12_RESOURCE_STATE_[COMMON|PRESENT], Missing State: 0x4: D3D12_RESOURCE_STATE_RENDER_TARGET. [ EXECUTION ERROR #538: INVALID_SUBRESOURCE_STATE]
` 
My computer is AMD radeon GPU and some people say use `WGPU_BACKEND= vulkan` can fix this problem but I don't know how to do that.

**Comment by bugsweeper** [comment]
If you are on Windows run
`set "WGPU_BACKEND=vulkan" && cargo run --example 2d_shapes`

**Reply by DasLixou** [reply]
For powershell, `$env:WGPU_BACKEND="vk"; cargo run --example 2d_shapes`.
Related wgpu issue: https://github.com/gfx-rs/wgpu/issues/4247

**Reply by FALLfield** [reply]
Got it， I‘ ll try it later!

**Comment by kudorgyozo** [comment]
How can I just set it to be vulkan without env variable?

**Reply by JMS55** [reply]
https://dev-docs.bevyengine.org/bevy/render/settings/struct.WgpuSettings.html#structfield.backends

**Reply by kudorgyozo** [reply]
> How can I just set it to be vulkan without env variable?

```
App::new()
        .add_plugins(DefaultPlugins.set(bevy::render::RenderPlugin {
            render_creation: bevy::render::settings::RenderCreation::Automatic(
                bevy::render::settings::WgpuSettings {
                    backends: Some(bevy::render::settings::Backends::VULKAN),
                    ..default()
                },
            ),
            ..default()
        }))
        .run();
```

**Reply by trash340** [reply]
Yup
